### DISCUSSION TITLE
mongoose default schema values are not inserted

### DISCUSSION THREAD

**Discussion by nikstaaar** [neuefische/questions#220] [Web] [answered]
Hi, when a new user logs into my app with github via nextauth, their username and e-mail gets saved in my mongodb, however the default values i built into my user schema don't get saved into mongodb. 

how my schema looks.

<img width="722" alt="Screenshot 2022-10-06 at 09 58 19" src="https://user-images.githubusercontent.com/104061897/194252181-6aa2cd76-1728-4f53-ac38-9762c0b694f5.png">

how my saved data looks:

<img width="979" alt="Screenshot 2022-10-06 at 09 57 52" src="https://user-images.githubusercontent.com/104061897/194251818-b1d96cd5-a909-4046-b4c8-38575ac09e2b.png">

https://github.com/nikstaaar/capstone-project/pull/15

I would greatly appreciate some ideas on how to fix this.

**Comment by nikstaaar** [comment]
ok i think I misunderstood how the mongodb adapter for nextauth works. the user schema isn't needed at all. I deleted it and the user data still gets saved.

**Comment by mbosselmann** [comment, marked as answer]
Hi Niklas,
i think the problem is that your user model of mongoose is not linked to your Next Auth functionality and the MongoDBAdapter. As far as I can see, you followed the [Next Auth documentation](https://next-auth.js.org/adapters/mongodb) for connecting NextAuth to MongoDB, which provides a special adapter for this.
To learn how you can implement a user model, you can read more about [models for Next Auth also in their docs](https://next-auth.js.org/adapters/models). There is a section on how you can customize the data to be stored in the database.
Feel free to try this out and report back if it works. :)

**Reply by nikstaaar** [reply]
I made it work with the instructions from the documentation! thank you 👍

**Reply by Septem-7** [reply]
Hey @nikstaaar I ran into the same problem. I want to extend the user schema (add roles and user name) when signing in with the email provider using the mongodb adapter but it doesn't work. Can you explain to me how you made it work? If I get it right you defined extra variables via mongoose and added them to the profile, right? Thanks a lot in advance.

**Reply by nikstaaar** [reply]
hi, you can take a look at my nextauth page where I added the github provider. 

https://github.com/nikstaaar/capstone-project/blob/main/src/pages/api/auth/%5B...nextauth%5D.js

at the bottom you can see I returned an object within the provider function with all the data I want in the model. When a user is created with the mongoDB adapter it uses that object instead of mongoose. 

hope this helps.

**Reply by Septem-7** [reply]
Thanks for your quick response, but it does not seem to work with the email provider and the mongodb adapter.
